Empty {} snakemake.yaml causes 500 error

**Bug Description**
The dataview reproduce endpoint (GET /api/dataview/workflow/reproduce/{workspace_id}/{unique_id}) returns an unhandled 500 error when a downloaded snakemake.yaml file contains an empty YAML mapping ({}). The AssertionError in SmkConfigReader.read() is not caught by any upstream exception handler, resulting in a raw internal server error instead of a meaningful response.

**Steps to Reproduce**
1. Navigate to a public or dataview experiment ecf9ab15 (workspace 121)
2. Trigger the reproduce endpoint (e.g., open the experiment in the dataview UI)
3. The system downloads all 41 output files from S3 bucket optinist-user-11-XXXXXXX successfully
4. The system attempts to read snakemake.yaml to determine input files for download
5. The endpoint returns HTTP 500

**Expected Behavior**
If experiment data is corrupt or incomplete, the endpoint should return a meaningful error response (e.g., 503 with a message like "Experiment data is incomplete or corrupt") rather than an unhandled exception.

**Actual Behavior**
An AssertionError propagates uncaught through the middleware stack, producing a 500 response. The assertion fires because yaml.safe_load returns {} for the empty config file, and assert {} is falsy in Python. The exception path is:
  - dataview.py:426 → _ensure_experiment_downloaded
  - remote_storage_controller.py:769 → SmkUtils.get_datatypes_inputs()
  - smk_utils.py:90 → SmkConfigReader.read()
  - snakemake_reader.py:51 → assert config fails
 
**Environment**
Platform: ECS (subscr-optinist-cloud-taskdef)

**Priority**
Classify the severity of this bug as [Low, Medium, High, Critical].
Low — The bug affects a single experiment with corrupt data. Other experiments are unaffected. Users see a generic 500 error but no data loss or security risk is involved.

**Screenshots/Logs**
 ```
2026-03-02 18:51:37,276 INFO: "GET /api/dataview/workflow/reproduce/121/ecf9ab15 HTTP/1.1" 500
AssertionError: Invalid config yaml file: [/app/studio_data/output/121/ecf9ab15/snakemake.yaml] [{}]
```

**Additional Comments**
- [ ] Data corruption investigation — It is currently unclear how the empty snakemake.yaml ended up in S3. The file exists and downloads successfully but contains no rules. Possible causes include a failed or interrupted experiment save, a race condition during upload, or an external modification. The S3 object metadata and version history (if versioning is enabled) should be examined to determine when and how the file was written in this state.
- [ ] Error handling improvement — Regardless of the data corruption root cause, the code path should handle invalid config gracefully. The _ensure_experiment_downloaded function catches RemoteExperimentNotFoundError and RemoteStorageLockError but does not catch AssertionError or a more specific invalid-config exception. Adding handling here would convert the raw 500 into a meaningful response for the user.

--- a/studio/app/common/core/storage/remote_storage_controller.py
+++ b/studio/app/common/core/storage/remote_storage_controller.py
@@ -766,11 +766,16 @@ class RemoteStorageController(BaseRemoteStorageController):
             # Only download input data and mark success for full syncs
             if is_full_sync:
                 # Download the input data related to the experiment data as well.
-                input_filenames = SmkUtils.get_datatypes_inputs(
-                    workspace_id, unique_id, apply_basename=True
-                )
-                for input_filename in input_filenames:
-                    await self.download_input_data(workspace_id, input_filename)
+                try:
+                    input_filenames = SmkUtils.get_datatypes_inputs(
+                        workspace_id, unique_id, apply_basename=True
+                    )
+                    for input_filename in input_filenames:
+                        await self.download_input_data(workspace_id, input_filename)
+                except (AssertionError, KeyError) as e:
+                    # snakemake.yaml may be empty or missing required keys
+                    logger.warning(f"snakemake yaml read error: {e}")
+                    pass
 
                 # update sync status file
                 RemoteSyncStatusFileUtil.create_sync_status_file_for_success(

